### PR TITLE
Added support for building the janusgraph dockerimage behind a proxy

### PR DESCRIPTION
**Pull request by StefWin**
Thank you for contributing to JanusGraph.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pluradj** [comment]
@StefWin thanks for the contribution! JanusGraph requires a [signed CLA](https://github.com/JanusGraph/janusgraph/blob/master/CONTRIBUTING.md#contributing-to-janusgraph) for contributions. Once that is filed, the CLA bot will update the label on this PR. Also we're in the process of a release, so if this PR is accepted, it won't be merged until after the release is complete. Thanks again.

**Comment by mbrukman** [comment]
@StefWin — thank you for signing the ICLA! I am adding you to our CLA signers list so you'll be able to contribute to JanusGraph soon. A few questions for you; I will follow-up by email.

**Comment by pluradj** [comment]
any updates on the CLA @StefWin @mbrukman?

**Comment by mbrukman** [comment]
@pluradj — CLA is being added in https://github.com/JanusGraph/legal/pull/94; please feel free to review & merge.

**Comment by mbrukman** [comment]
@pluradj, @StefWin — the CLA has been taken care of and @janusgraph-bot has updated the tags on this PR accordingly. You're all set!

**Comment by chupman** [comment]
non-binding LGTM,

I forked built the dist with  `mvn clean install -Pjanusgraph-release -Dgpg.skip=true -DskipTests=true`
set a bogus http_proxy and ran 
`mvn docker:build -Pjanusgraph-docker -pl janusgraph-dist`  
The build failed to pull down the release files for apt. I then unset the ENV and was able to run the docker build command successfully.

If it helps with the real review here's the [link](https://docs.docker.com/engine/reference/builder/#predefined-args) to http_proxy being a predefined ARG for dockerfiles.

File: janusgraph-dist/pom.xml [hunk 1]
```diff
@@ -569,6 +569,7 @@
                                 <dockerDirectory>${project.basedir}/janusgraph-dist-hadoop-2</dockerDirectory>
                                 <buildArgs>
                                     <server_zip>target/janusgraph-${project.version}-hadoop2.zip</server_zip>
+                                    <http_proxy>${env.http_proxy}</http_proxy>
```
Comment (chupman) [inline janusgraph-dist/pom.xml]: In BUILDING.md under the [Building Docker Image for JanusGraph Gremlin Server](https://github.com/JanusGraph/janusgraph/blob/master/BUILDING.md#building-docker-image-for-janusgraph-gremlin-server) section please add something along the lines of:


> If you are building the Docker image behind a proxy please set an environment variable for either `http_proxy` or `https_proxy` accordingly.

**Comment by farodin91** [comment]
@sjudeng We can close this PR or?

**Comment by chupman** [comment]
@farodin91 @sjudeng I'm fine with merging it as is and adding the additional https_proxy env and documentation as a follow-up. Any objections?

**Comment by porunov** [comment]
@chupman Looks like there are no objections. I think it is OK to merge it.

**Comment by farodin91** [comment]
@chupman you can merge from my side.

**Comment by chupman** [comment]
Open comments were addressed with a follow-up commit in #1582 along with this commit. Closing as a duplicate.

**Pull request closed without merging**